### PR TITLE
starting review

### PR DESCRIPTION
**Pull request by jspaaks**

### CODE REVIEW

**Comment by jspaaks** [comment]
@sverhoeven could you give me some kind of write access such that I can annoate the code as a reviewer?

**Comment by sverhoeven** [comment]
I have invited you as a writer for this repo some time ago, could you check your mailbox?

**Comment by jspaaks** [comment]
OK great.

So I was hoping for better comment capabilities, but that doesnt seem to be possible. Below I'm compiling a list of what would have been comments. For typos and such, I'll just make commits.

First off: great job Stefan, really nice piece.

(So now the comments):

1. [ ] for the mathematical symbols, I'd prefer to use ``tolerance`` instead of ``epsilon``
1. [ ] there is some commented-out code that mentions Bisection methods, but I believe that is a different method that Newton-Raphson.
1. [ ] for each code snippet, consider adding its file name as a comment at the top, this is useful since some snippets points to each other's content
1. [ ] Instead of ``h``, I would prefer ``delta_x``
1. [ ] ``guess`` could be ``x_guess``
1. [ ] the comment ``// x(i+1) = x(i) - f(x) / f'(x)11 is confusing (to me)
1. [ ] where it says ``<<algorithm>>`` perhaps we can use ``#include`` instead?
1. [ ] the content of ``namespace rootfinding`` could use 2 indents

File: README.md [hunk 1]
```diff
@@ -279,15 +279,15 @@ Python packages can be installed using `pip` from the [Python Package Index](htt
...
+Python can call functions in a C++ library if its functions use [Python.h datatypes](https://docs.python.org/3.7/extending/index.html). This requires a lot of boilerplate and conversions, several tools are out there that make the boilerplate/conversions much simpler. The tool we chose to use is [pybind11](https://github.com/pybind/pybind11) as it is currently (May 2019) actively maintained and is a header-only library.
 
 To use pybind11, it must installed with pip
 
 ```{.awk #pip-pybind11}
 pip install pybind11
 ```
 
-Pybind11 requires a bindings (PYBIND11_MODULE macro) to expose C++ constants/functions/enumerations/classes to Python. The bindings can be compiled to a shared library (eg. pybubble.so) which can be imported into Python.
+Pybind11 requires a bindings macro (PYBIND11_MODULE) to expose C++ constants/functions/enumerations/classes to Python. The bindings can be compiled to a shared library (eg. pybubble.so) which can be imported into Python.
```
Comment (jspaaks) [inline README.md]: not sure this is what you meant but ok

File: README.md [hunk 2]
```diff
@@ -313,7 +313,7 @@ PYBIND11_MODULE(newtonraphsonpy, m) {
 
 Compile with
 
-```{.shell #build-py}
+```{.awk #build-py}
```
Comment (jspaaks) [inline README.md]: not sure why this needs to be awk, further up ([244](https://github.com/NLESC-JCER/cpp2wasm/blame/master/README.md#L244)) there is another instance of .shell, but that's not used so maybe thats why it passes

**Comment by jspaaks** [comment]
@sverhoeven, I'm available to go over my comments together if you want

File: README.md (hunk 1 above)
Comment (sverhoeven) [reply README.md]: Put macro in own sentence in https://github.com/NLESC-JCER/cpp2wasm/pull/17/commits/f68628c6165ab16bbc557bd5b3f62f166e87f2f4

File: README.md (hunk 2 above)
Comment (sverhoeven) [reply README.md]: entangled does not know about [shell](https://github.com/entangled/entangled/blob/master/src/Config.hs#L37-L59), so I used awk as an alternative. The other `shell` commands are not entangled so can still use the `shell` naming. Made consistent in https://github.com/NLESC-JCER/cpp2wasm/pull/17/commits/129577d4eadb5e7fa6a8adb03295a7332104fe27

**Comment by sverhoeven** [comment]
Resolving comments in https://github.com/NLESC-JCER/cpp2wasm/pull/17 PR

**Comment by sverhoeven** [comment]
Link to original algorithm C++ source

**Pull request merged**